### ISSUE TITLE
PROPOSAL: Make argo accessor not use adjusted parameters

### ISSUE THREAD

**Issue by j-harbin** [dankelley/oce#1711]
As discussed with @dankelley in a vftf, we noticed in argo.R lines 156-158 the accessor is grabbing the adjusted data. We went back and both on if this should be the case. @richardsc , what are your thoughts on this?

**Comment by richardsc** [comment]
Ooof ... I'm really not too sure how to handle this. It seems obvious that if "*_ADJUSTED" exists that's the one that should be used, but as I recall there are lots of examples where the ADJUSTED fields are all NAs, right?

In any case, it doesn't seem to be documented all that clearly (especially if you found out by actually looking at the source code rather than the help page), so that's something that could be cleaned up.

One idea I thought of is if the object has a `metadata` field called e.g. `useAdjusted`, which could be TRUE or FALSE, which would dictate the behaviour, maybe with an accompanying function, e.g.

```r
library(oce)
data(argo)
argo_adj <- useAdjusted(argo)
print(argo_adj[['useAdjusted']])
TRUE
```

**Comment by dankelley** [comment]
Hm.  @richardsc you are a genius.  What `useAdjusted` should do is simply to set a flag.  And that flag would get used by `[[`.

The whole topic is tricky:
1. From what I have seen, adjusted S and T seem to be straightforward: they just copy the original values unless they do not like the flags (in which case they insert 'NA').  But I do not have any documentation (that I recall) telling me that for sure.  And I'm not even sure who "they" are ... the PIs? ... an algorithm at the dac?
2. For bgc type things, though, it is a mess.  It seems that if they adjust a variable, then they put "_ADJUSTED" fields in *all* the fields, but things that are not actually examined or processed (or something) get set to NA

@j-harbin and I have cooked up a fairly detailed scheme (in the docs for `argoFloats::useAdjusted`) to try to handle these conditions.

I wonder if the 3 of us could do a quick zoom meeting to discuss this.  I just got out of a meeting and have time now.

**Comment by dankelley** [comment]
Oh, and both the `useAdjusted()` function and the flag it stores in the metadata could also supply the variable name, so the user could use (say) adjusted oxygen, but unadjusted salinity.  (If my theory about adjusted S and T is right, I would prefer not to use them, because I would rather use flags myself, being more cautious than the "normal" or less cautious, depending on my needs.)

**Comment by dankelley** [comment]
I am making a new branch, `"issue1711"`, in which I will try out @richardsc's proposal that
* `useAdjusted()` should only alter something in the `metadata` slot, relying on `[[` to then obey the setting,

instead of the existing

* `useAdjusted()` renames data (and flags, and units, and originalNames), leaving `[[` alone.

**Comment by dankelley** [comment]
This is done in commit bba0967897fff4ed343e71d14a99f6071c3608bf of the "issue1711-useAdjusted" branch.  I called the function `selectAdjusted()`, to avoid naming conficts with the [argoFloats package](https://github.com/argoCanada/argoFloats) definition of `useAdjusted()`.

NOTES:

1. Branch  "issue1711-useAdjusted" has version number bumped to 1.3.0 (from 1.2.x) because of the changes to `read.argo()` and `data(argo)`.  I forgot to do that when I did those updates, a while back.  This bump will be taken into account by the argoFloats package.

2. I plan to merge "issue1711-useAdjusted" into "develop" at 1200h tomorrow.  After that, I will adjust `argoFloats::useAdjusted()` to use `oce::preferAdjusted()`.
